src/_pytest/main.py

```diff
@@ -383,7 +386,9 @@ def __init__(self, config):
         self._initialpaths = frozenset()
         # Keep track of any collected nodes in here, so we don't duplicate fixtures
         self._node_cache = {}
-        self._bestrelpathcache = _bestrelpath_cache(config.rootdir)
+        self._bestrelpathcache = _bestrelpath_cache(
```

Comment:
Not sure if that's even possible, but does changing `_bestrelpath_cache` to inherit from `Dict[str, str]` instead of `dict` work? If so, then this could be avoided I think.

Reply:
Does not work for https://github.com/blueyed/pytest/blob/1b3c0e6d5c078dadd11dfb2a109dc330742ac75c/src/_pytest/main.py#L408.

Reply:
I think _bestrelpath_cache should be dropped, instead of being fixed further..
(using generic caching around pathlib likely)

Reply:
It does not work because `__init__` is untyped. Adding `-> None` to `__init__` would be sufficient to get it working, I think. Although you would then need to type the couple empty collections `self._initialpaths` and `self._node_cache`, if you know what their types are supposed to be.

Reply:
I'd rather leave it as-is then.